modules/core/SolidTorus.cpp

```diff
@@ -0,0 +1,53 @@
+#include <macroses.h>
+#include "SolidTorus.h"
+#include "PrimTriangle.h"
+
+rt::CSolidTorus::CSolidTorus(const rt::ptr_shader_t &pShader, const Vec3f &origin, float r1, float r2, int nSides,
```

Comment:
Here we can distinguish the torus sides and segments.
If we put a torus at the table and cut it as a cake to pieces - these part are called _segments_
If we take one such piece - it will link like a deformed pipe of cylinder. The number of sides of such cylinder is the number of _sides_. usually `int nSegments = 2 * nSides` I think that the loops for i and j should be adjusted, that one of them is iterated till 2 * nSides...